### ISSUE TITLE
Custom URLs in galleries without schema creates relative links

### ISSUE THREAD

**Issue by mmtr** [godaddy/coblocks#747]
**Describe the bug**
The custom URLs entered in the gallery images are not being sanitized / validated, so entering an URL without the schema (`http(s)://`) will create a relative URL.

**To Reproduce**
1. Insert a gallery block such as Masonry.
1. Add some images to the gallery.
1. Under "Link Settings" in the Block Settings, choose Custom URL.
1. Enter an address without http:// or https:// (i.e. `godaddy.com`).
1. Publish the page/post.
1. Visit the published post/page.

**Expected behavior**
To be taken to the external website upon clicking the gallery image, but instead you're taken to an internal path (i.e. `example.com/2019/08/17/my-post/godaddy.com`).

**Isolating the problem:**
-   [x] This bug happens with no other plugins activated
-   [x] This bug happens with a default WordPress theme active
-   [x] This bug happens **without** the Gutenberg plugin active
-   [x] I can reproduce this bug consistently using the steps above

**WordPress Version**
5.2

**Gutenberg Version**
6.3.0

**Comment by kwight** [comment]
> entering an URL without the schema (http(s)://) will create a relative URL

Hm, I'm not sure I'd consider this a bug; dot characters are valid in URL paths (think of versioned API URLs), so the presence of a dot doesn't mean we aren't referring to a relative path. Maybe the "fix" is a description in the Link Settings section specifying that the schema needs to be explicit for links to other domains?

**Comment by mmtr** [comment]
> the presence of a dot doesn't mean we aren't referring to a relative path.

Yeah, I'm not suggesting that. I only think a user will expect an absolute URL in most of the scenarios. Note that absolute URLs will work for linking to internal pages too (the input allows to search for any blog page/post and it'll use the absolute URL). So maybe it'll be simpler (from a user perspective) if the plugin assumes all the URLs are absolute and append the schema when not entered.

**Comment by richtabor** [comment]
Appreciate the report. Does the core Gallery Block behave the same way?

**Comment by mmtr** [comment]
@richtabor AFAIK the core Gallery Block doesn't allow to link images to custom URLs.

**Comment by mmtr** [comment]
🤔 Actually, the core image block does allow it and has the same behavior. Reported there as well: https://github.com/WordPress/gutenberg/issues/17155

Are the CoBlocks galleries using the same tool for linking the images?

**Comment by richtabor** [comment]
> 🤔 Actually, the core image block does allow it and has the same behavior. Reported there as well: [WordPress/gutenberg#17155](https://github.com/WordPress/gutenberg/issues/17155)
> 
> Are the CoBlocks galleries using the same tool for linking the images?

Ahhhh I see now. We are for the most part, but I've added the custom URL functionality as that doesn't exist in core.
